src/cmd/singularity/cli/actions.go

```diff
@@ -466,7 +464,56 @@ func execStarter(cobraCmd *cobra.Command, image string, args []string, name stri
 		}
 	}
 
-	if err := exec.Pipe(starter, []string{procname}, Env, configData); err != nil {
-		sylog.Fatalf("%s", err)
+	if engineConfig.GetInstance() {
+		stdout, stderr, err := instance.SetLogFile(name)
+		if err != nil {
+			sylog.Fatalf("failed to create instance log files: %s", err)
+		}
+
+		pipefd, err := exec.SetPipe(configData)
+		if err != nil {
+			sylog.Fatalf("failed to set pipe communication: %s", err)
+		}
+
+		cmd := osexec.Command(starter, "")
+		start, err := stderr.Seek(0, os.SEEK_END)
+		if err != nil {
+			sylog.Warningf("failed to get standard error stream offset: %s", err)
+		}
+
+		cmd.Stdout = stdout
+		cmd.Stderr = stderr
+		cmd.Env = append(Env, pipefd)
+		cmd.Args[0] = procname
+
+		cmdErr := cmd.Run()
```

Comment:
Is intended to use `cmd.Run()` since `Run()` method blocks until `cmd` exits (either successfully or not-successfully)?

Reply:
Yes, instance always returns, so `cmd.Run()` will block until instance start or fail to provide information about success/failure